App/templates/App/index.html

```diff
@@ -19,7 +19,7 @@
 	    	{{error}}
 	    {% endif %}
 	    {% if username %}
-	    	Username = {{username}} <br>
+	    	Username = {{username | safe}} <br>
```

Comment:
This turns off escaping. There is no reason to do that and it will result in some usernames looking wrong (if they contain a HTML entity). It could also be a security hole although here it probably won't be since MediaWiki disallows `<>` in usernames.

Default autoescaping works well when you just want to display text. Where it doesn't work is when you want use a variable in an URL or a Javascript variable or any other context where the escaping/encoding rules are different from that of HTML.

Reply:
True, I understood not this thing in the first instance. Marking it safe is not the solution. So I have handled this now, using urlencoding. Now, the code works for all possible usernames else displays the error msg.